lnwallet/channel.go

```diff
@@ -6916,16 +6913,54 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64,
 	// Ensure the fee rate doesn't dip below the fee floor.
 	_, weight := lc.availableBalance()
 	maxFeeRate := maxFee / (float64(weight) / 1000)
-	feeRate := chainfee.SatPerKWeight(
+	return chainfee.SatPerKWeight(
 		math.Max(maxFeeRate, float64(chainfee.FeePerKwFloor)),
 	)
+}
+
+// IdealCommitFeeRate uses the current network fee, the minimum relay fee,
+// maximum fee allocation and anchor channel commitment fee rate to determine
+// the ideal fee to be used for the commitments of the channel.
+func (lc *LightningChannel) IdealCommitFeeRate(netFee, minRelayFee,
+	maxAnchorCommitFeeRate chainfee.SatPerKWeight,
+	maxFeeAlloc float64) chainfee.SatPerKWeight {
+
+	maxFee := lc.maxFeeRate(maxFeeAlloc)
+
+	var commitFee chainfee.SatPerKWeight
+
+	// If the channel does not have anchor outputs then we cap the fee at
+	// the max fee allocation otherwise we cap the fee at the max anchor
+	// commit fee rate.
+	switch lc.channelState.ChanType.HasAnchors() {
+	case false:
+		commitFee = chainfee.SatPerKWeight(
+			math.Min(float64(netFee), float64(maxFee)),
+		)
+
+	case true:
+		commitFee = chainfee.SatPerKWeight(
+			math.Min(
+				float64(netFee),
+				float64(maxAnchorCommitFeeRate),
+			),
+		)
+	}
+
+	// If the commitment fee is below the minimum relay fee then we bump it
+	// to be equal to the minimum relay fee.
+	if commitFee < minRelayFee {
+		commitFee = minRelayFee
+	}
 
-	// Cap anchor fee rates.
-	if lc.channelState.ChanType.HasAnchors() && feeRate > maxAnchorFeeRate {
-		return maxAnchorFeeRate
+	// In the case were the commitment fee is more than the maximum fee
+	// we are able to pay then we set the new commitment fee equal to this
+	// max fee.
+	if commitFee > maxFee {
```

Comment:
if this check is switched with the min relay fee check then it wont do anything since the switch above it caps commit fee at max fee. But yeah, I will change this up a bit depending on the response to my questions [here](https://github.com/lightningnetwork/lnd/pull/5483#issuecomment-898362068)

Reply:
Maybe my comment wasn't clear.

Currently:
- switch statement caps at maxfee
- happens to be < minRelayFee, set to minRelayFee
- now > maxFee, set to maxFee (which is < minRelayFee)

My suggestion:
- switch statement caps at maxfee
- won't be > maxFee, do nothing
- may be < minRelayFee, set to minRelayFee

Unless I'm missing something?

Reply:
see latest update :)